pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -71,3 +85,81 @@ func (p RESTStorageProvider) v1alpha1Storage(apiResourceConfigSource serverstora
 func (p RESTStorageProvider) GroupName() string {
 	return flowcontrol.GroupName
 }
+
+func (p RESTStorageProvider) PostStartHook() (string, genericapiserver.PostStartHookFunc, error) {
+	return PostStartHookName, func(hookContext genericapiserver.PostStartHookContext) error {
+		flowcontrolClientSet := flowcontrolclient.NewForConfigOrDie(hookContext.LoopbackClientConfig)
+		return wait.PollImmediateUntil(time.Minute, func() (bool, error) {
+			// ensure the pre-defined objects if the exempt flow-schema is present.
+			shouldEnsure, err := shouldEnsureAllPredefined(flowcontrolClientSet)
```

Comment:
once in a controller, I think you'll see the races.  Things like your presence signal here isn't "only one fires". I'm ok merging it like this, but I suspect you actually want to try to create the required objects and if any are able to be created then you create the rest.